init/eessi_environment_variables

```diff
@@ -10,7 +10,7 @@ function error() {
 function show_msg {
   # only echo msg if EESSI_SILENT is unset
   msg=$1
-  if [[ -z ${EESSI_SILENT+x} ]]; then
+  if [[ -z ${EESSI_SILENT:+x} ]]; then
```

Comment:
I don't think this is what we want. According to chatgpt:

"To summarize:

- If you want to check specifically whether `EESSI_SILENT` is both set and not empty, you would use ${EESSI_SILENT:+x}.
- If you only want to check whether `EESSI_SILENT` is set, regardless of its value (empty or not), you use ${EESSI_SILENT+x}."
 
We only want to know if it's set. If it's set (and empty), we still don't want to print output.

Reply:
Right. I find it non-intuitive. Maybe we should opt for more easily to understand logic.